sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Filter.java

```diff
@@ -207,43 +166,54 @@ public void populateDisplayData(DisplayData.Builder builder) {
    * <p>See also {@link #byPredicate}, which returns elements
    * that satisfy the given predicate.
    */
-  public static <T extends Comparable<T>> ParDo.Bound<T, T> greaterThanEq(final T value) {
-    return ParDo.named("Filter.greaterThanEq").of(new DoFn<T, T>() {
-      @Override
-      public void processElement(ProcessContext c) {
-        if (c.element().compareTo(value) >= 0) {
-          c.output(c.element());
-        }
-      }
-
+  public static <T extends Comparable<T>> Filter<T> greaterThanEq(final T value) {
+    return byPredicate(new SerializableFunction<T, Boolean>() {
       @Override
-      public void populateDisplayData(DisplayData.Builder builder) {
-        super.populateDisplayData(builder);
-        Filter.populateDisplayData(builder, String.format("x ≥ %s", value));
+      public Boolean apply(T input) {
+        return input.compareTo(value) >= 0;
       }
-    });
+    }).named("Filter.greaterThanEq").described(String.format("x ≥ %s", value));
   }
 
   ///////////////////////////////////////////////////////////////////////////////
 
+  private String name = "Filter";
```

Comment:
The way we encourage is this:

```java
myPCollection.apply("Name For Filter", Filter.by(...))
```

Because if a `PTransform` instance is used twice, we want it to have two stable, unique names. Previously the name provided by `named` would get a generated suffix, but that is not stable.